addon/components/bs-datetimepicker.js

```diff
@@ -73,113 +131,30 @@ var bsDateTimePickerComponent = Ember.Component.extend({
-    }
-  }),
-
-  _disabledDatesObserver: Ember.observer('disabledDates', function() {
-    this.get('bsDateTimePicker').disabledDates(this.get('disabledDates'));
-  }),
-
-  _enabledDatesObserver: Ember.observer('enabledDates', function() {
-    this.get('bsDateTimePicker').enabledDates(this.get('enabledDates'));
-  }),
+  _destroyDatepicker: on('willDestroyElement', function() {
+    this.removeObserver('disabled');
+    this.removeObserver('open');
+    this.removeObserver('minDate');
+    this.removeObserver('maxDate');
+    this.removeObserver('disabledDates');
+    this.removeObserver('enabledDates');
+    this.removeObserver('date');
 
-  _dateObserver: Ember.observer('date', function() {
-    var bsDateTimePickerFn = this.get('bsDateTimePicker');
-
-    if (this.get('date') === undefined) {
-      bsDateTimePickerFn.date(null);
-    } else {
-      bsDateTimePickerFn.date(this.get('date'));
-    }
-
-  }),
-
-  _destroyDatepicker: Ember.on('willDestroyElement', function() {
     this.get('bsDateTimePicker').destroy();
   }),
 
-  _buildConfig: function() {
+
+  _buildConfig() {
     var config = {};
     for (var i = 0; i < isDatetimepickerConfigKeys.length; i++) {
       config[isDatetimepickerConfigKeys[i]] = this.get(isDatetimepickerConfigKeys[i]);
     }
 
     return config;
-  },
-
-  inputGroupClass: Ember.computed(function() {
-    if (!this.get('noIcon')) {
-      return 'input-group';
-    }
-  }),
-
-  /**
-    Exposing the textField properties.
-    Every property beginning with 'textField' will be exposed to the TextField view.
-
-    ```handlebars
-    {{bs-datetimepicker textFieldPlaceholder='Date'}}
-    ```
-    'textFieldPlaceholder' will be exposed to the TextField as 'placeholder' property.
-  */
-  setUnknownProperty: function(key, value) {
-    var prop;
-    var ckey;
-
-    if (key.indexOf('textField') === 0) {
-      if (Ember.isNone(this.get('textFieldOptions'))) {
-        this.set('textFieldOptions', {});
-      }
-
-      if (Ember.IS_BINDING.test(key)) {
-        prop = key.substring(0, key.length-7);
-      } else {
-        prop = key.substring(0, key.length);
-      }
-
-      ckey = prop.substring(9);
-      ckey = ckey.charAt(0).toLowerCase() + ckey.substr(1);
-
-      if (Ember.isNone(this.get('textFieldOptions.' + prop))) {
-
-        this.set('textFieldOptions.' + prop, ckey);
-
-        Ember.defineProperty(this, prop, null, value);
-      }
-    } else {
-      if (Ember.platform.hasPropertyAccessors) {
-        Ember.defineProperty(this, key, null, value);
-      } else {
-        this[key] = value;
-      }
-    }
   }
 });
 
-var computedProps = Ember.A(['minDate', 'maxDate', 'disabledDates', 'enabledDates', 'dateIcon']);
-var newClassConfig = {};
 for (var i = 0; i < isDatetimepickerConfigKeys.length; i++) {
   if (!computedProps.contains(isDatetimepickerConfigKeys[i])) {
     newClassConfig[isDatetimepickerConfigKeys[i]] = datetimepickerDefaultConfig[isDatetimepickerConfigKeys[i]];
```

Comment:
We probably don't need to reopen since we are creating the addon here, it can be part of the original POJO passed to `Component.extend`